### ISSUE TITLE
Suggests for auto import only works if you have imported a library at least once somewhere else

### ISSUE THREAD

**Issue by oliverandrich** [microsoft/pylance-release#1283]
## Environment data

-   Language Server version: 2021.5.2
-   OS and version: macOS 11
-   Python version (& distribution if applicable, e.g. Anaconda): Python 3.9.5

## Expected behaviour

I expect the suggest for an auto import to work immediately and without having to import the library elsewhere.

## Actual behaviour

Currently, I don't get a suggestion for auto import (Screenshot 1) until I manually add the import somewhere. I can even remove it afterwards and still get suggestions for the autoimport (Screenshot 2). It seems PyLance needs some kind of warmup here?

![CleanShot 2021-05-14 at 12 05 03@2x](https://user-images.githubusercontent.com/758/118256639-02402700-b4ae-11eb-9635-7fc078433b44.png)

![CleanShot 2021-05-14 at 12 06 39@2x](https://user-images.githubusercontent.com/758/118256651-05d3ae00-b4ae-11eb-900a-4591869f4102.png)

## Logs

Enabling python.analysis.logLevel didn't create any meaningfull output on a suggest. I post the content anyway. Maybe it helps.

```
User belongs to experiment group 'pythonaacf'
User belongs to experiment group 'pythonJediLSP'
User belongs to experiment group 'pythonDiscoveryModuleWithoutWatcher'
User belongs to experiment group 'pythonTensorboardExperiment'
User belongs to experiment group 'pythonFindInterpreter'
User belongs to experiment group 'PythonPyTorchProfiler'
User belongs to experiment group 'CollectLSRequestTiming - experiment'
Python interpreter path: ./.venv/bin/python
Starting Pylance language server.
> conda --version
> /usr/local/bin/black --diff --quiet ~/Projekte/privat/bla/bla/routers/version.py.8df8aade3f37070484540e5c92926a2d.tmp
cwd: ~/Projekte/privat/bla
```

**Comment by jakebailey** [comment]
At the moment, this is expected, but if you enable the (work in progress) indexing setting, you should see this working.

```
"python.analysis.indexing": true
```

This indexes user and library code at startup, which is then used for auto-import completions and quick fixes.

You can also enable our insiders build, where this is currently enabled by default: #1287 623

**Comment by oliverandrich** [comment]
@jakebailey If I try to add this to my config, it tells me that it is an unknown configuration option. I also activated the insiders build without success.

**Comment by jakebailey** [comment]
The option is hidden; if it's present then it will be used. With trace logging enabled, you should be able to see indexing happening.

Are you saying that the insiders system didn't work (and you weren't upgraded), or that it didn't fix the issue? If insiders didn't get installed when you changed the setting, then that'd certainly be worth a bug.

**Comment by oliverandrich** [comment]
Okay, the suggestion does (mostly) work now. Nice to learn that there are also hidden options. 

Concerning the insiders system. I still have the same version of Pylance as before.

`[Info  - 18:46:32] Pylance language server 2021.5.2 (pyright df8b6b3c) starting`

**Comment by jakebailey** [comment]
Hm, curious. If you've set `"pylance.insidersChannel": "daily"`, it should automatically install the update and prompt you to reload; can you check in your extensions menu to see if it's waiting for a reload?

You can also try removing the setting and adding it again, which will force a recheck. Otherwise, if you could make a new issue for tracking this discrepancy, that's be great.

**Comment by oliverandrich** [comment]
I did uncomment and change the setting now various, but nothing happened. Now as I was about record a video of it, the update is incoming, and I was asked to reload. :)

**Comment by jakebailey** [comment]
Hah, of course.

Let us know how the indexing setting works for you; we're currently working on improving its performance and impact on other features (lots of auto-imports can make completions slow from the sheer number of items, for example), hence why it's not enabled by default.

I'm going to close this issue, as indexing helps and is the intended fix (though currently with caveats like #1055).
